src/main/scala/wdlTools/eval/stdlib/BaseStdlib.scala

```diff
@@ -1,197 +1,227 @@
   }
 
   // Array[String] read_lines(String|File)
   //
-  private def read_lines(args: Vector[V], text: TextSource): V_Array = {
+  // since: draft-1
+  // deprecation: beginning in draft-2, URI parameter is not supported
+  protected def read_lines(args: Vector[V], text: TextSource): V_Array = {
     val file = getWdlFile(args, text)
-    val content = iosp.readFile(file.value)
-    val lines = content.split("\n")
-    V_Array(lines.map(x => V_String(x)).toVector)
+    val content = iosp.readFile(file.value, text)
+    V_Array(content.getLines.map(x => V_String(x)).toVector)
   }
 
+  private val tsvConf = CsvConfiguration('\t', '"', QuotePolicy.WhenNeeded, Header.None)
+
   // Array[Array[String]] read_tsv(String|File)
   //
-  private def read_tsv(args: Vector[V], text: TextSource): V_Array = {
+  // since: draft-1
+  // deprecation: beginning in draft-2, URI parameter is not supported
+  protected def read_tsv(args: Vector[V], text: TextSource): V_Array = {
     val file = getWdlFile(args, text)
-    val content = iosp.readFile(file.value)
-    val lines: Vector[String] = content.split("\n").toVector
-    V_Array(lines.map { x =>
-      val words = x.split("\t").toVector
-      V_Array(words.map(V_String))
-    })
+    val content = iosp.readFile(file.value, text)
+    val reader = content.mkString.asCsvReader[Vector[String]](tsvConf)
+    V_Array(reader.map {
+      case Left(err) =>
+        throw new EvalException(s"Invalid tsv file ${file}: ${err}", text, docSourceUrl)
+      case Right(row) => V_Array(row.map(x => V_String(x)))
+    }.toVector)
   }
 
   // Map[String, String] read_map(String|File)
   //
-  private def read_map(args: Vector[V], text: TextSource): V_Map = {
+  // since: draft-1
+  // deprecation: beginning in draft-2, URI parameter is not supported
+  protected def read_map(args: Vector[V], text: TextSource): V_Map = {
     val file = getWdlFile(args, text)
-    val content = iosp.readFile(file.value)
-    val lines = content.split("\n")
-    V_Map(lines.map { x =>
-      val words = x.trim.split("\t")
-      if (words.length != 2)
-        throw new EvalException(s"read_tsv ${file}, line has ${words.length} words",
-                                text,
-                                docSourceUrl)
-      V_String(words(0)) -> V_String(words(1))
-    }.toMap)
+    val content = iosp.readFile(file.value, text)
+    val reader = content.mkString.asCsvReader[(String, String)](tsvConf)
+    V_Map(
+        reader
+          .map {
+            case Left(err) =>
+              throw new EvalException(s"Invalid tsv file ${file}: ${err}", text, docSourceUrl)
+            case Right((key, value)) => V_String(key) -> V_String(value)
+          }
+          .toVector
+          .toMap
+    )
   }
 
   // Object read_object(String|File)
-  private def read_object(args: Vector[V], text: TextSource): V_Object = {
+  //
+  // since: draft-1
+  // deprecation:
+  // * beginning in draft-2, URI parameter is not supported
```

Comment:
I didn't know that URLs (http addresses) were no longer supported. That's good to know, because it makes things more complex.